modmesh/app/euler1d.py

```diff
@@ -61,73 +66,315 @@ def load_app():
 class QuantityLine:
     ana: matplotlib.lines.Line2D = None
     num: matplotlib.lines.Line2D = None
+    axis: matplotlib.pyplot.axis = None
     name: str = ""
     unit: str = ""
 
+    def bind_axis(self, axis):
+        self.axis = axis
+
     def update(self, adata, ndata):
         self.ana.set_ydata(adata.copy())
         self.num.set_ydata(ndata.copy())
+        self.axis.relim()
+        self.axis.autoscale_view()
         self.ana.figure.canvas.draw()
         self.num.figure.canvas.draw()
 
 
-class Plot:
-    def __init__(self, figsize=(15, 10)):
-        self.fig = Figure(figsize=figsize)
-        self.canvas = FigureCanvas(self.fig)
-        self.ax = self.canvas.figure.subplots()
-        self.ax.set_ylim(-0.2, 1.2)
-        self.ax.grid()
+class PlotManager(QWidget):
```

Comment:
By doing this (and the module-level importing pyside6), we are requiring pyside6 in this app.  The PR and check-in log should make this new requirement explicit.

Reply:
@j8xixo12 could you please address this point as well?

Reply:
I just updated PR message to remind `PySide6` is mandatory now.
Since `PySide6` is mandatory, the `MODMESH_USE_PYSIDE` seems is unnecessary, but it seems some issue need to fix https://github.com/solvcon/modmesh/blob/965aff9ca52ab3f00f80e4ce32aa48adbfbf2038/modmesh/app/euler1d.py#L153

I’ll try to reslove it.

Reply:
If the issue does not get in your way of this PR, it's OK to use another PR to fix it.